src/ProtectedProxies/MinimalProtectedProxyFactory.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: UNLICENSED
+// (c) SphereX 2023 Terms&Conditions
+
+pragma solidity ^0.8.0;
+
+import {SphereXProtected} from "../SphereXProtected.sol";
+import {SpherexProtetedMinimalProxy} from "./ProtectedMinimalProxy.sol";
+
+// TODO: add event for deploying
+
+/**
+ * @title A factory contract that deploys the SpherexProtetedMinimalProxy contracts.
+ */
+contract MinimalProtectedProxyFactory is SphereXProtected {
```

Comment:
the spherexprotected has more functionality than we need here. please add another layer of abstraction spherexstorage\spherexconfiguration something like that and make this one inherit directly from it